src/System.Net.Ping/tests/FunctionalTests/PingTest.cs

```diff
@@ -845,5 +845,14 @@ public void Ping_TimedOutSync_Success()
             PingReply reply = sender.Send(TestSettings.UnreachableAddress);
             Assert.Equal(IPStatus.TimedOut, reply.Status);
         }
-     }
+
+        [Fact]
+        [OuterLoop]
+        public async Task PingAsync_TimedOutSync_Success()
```

Comment:
Do we have sufficient tests for the older async `Ping.SendAsync()` API?  Perhaps we should have one more additional "async" test about "TimedOut" using the other event-based async Ping API.

Reply:
Ultimately I think we could do it similarly to HTTP testing... write tests in a base class, and have three implementations that shuffle to sync/EAP/TAP.